iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -25,57 +25,73 @@ public class IotHubDeviceClient : IDisposable
         }
 
-        /// <summary>
-        /// Creates a disposable DeviceClient using AMQP transport from the specified connection string
-        /// </summary>
-        /// <param name="connectionString">Connection string for the IoT hub (including DeviceId)</param>
-        /// <param name="options">The options that allow configuration of the device client instance during initialization.</param>
-        /// <returns>A disposable DeviceClient instance</returns>
-        /// <exception cref="ArgumentNullException"><paramref name="connectionString"/>, IoT hub host name or device Id is null.</exception>
-        /// <exception cref="ArgumentException"><paramref name="connectionString"/>, IoT hub host name or device Id are an empty string or consist only of white-space characters.</exception>
-        /// <exception cref="ArgumentException">Neither shared access key nor shared access signature were presented for authentication.</exception>
-        /// <exception cref="ArgumentException">Either shared access key or shared access signature where presented together with X509 certificates for authentication.</exception>
-        public static IotHubDeviceClient CreateFromConnectionString(string connectionString, IotHubClientOptions options = default)
+        private IotHubDeviceClient(IotHubConnectionCredentials iotHubConnectionCredentials, IotHubClientOptions options)
         {
-            Argument.AssertNotNullOrWhiteSpace(connectionString, nameof(connectionString));
-            return Create(() => ClientFactory.CreateFromConnectionString(connectionString, options));
-        }
+            if (iotHubConnectionCredentials.ModuleId != null)
+            {
+                throw new ArgumentException("A module Id was specified in the connection string - please use IotHubModuleClient for modules.");
+            }
 
-        private static IotHubDeviceClient Create(Func<InternalClient> internalClientCreator)
-        {
-            return new IotHubDeviceClient(internalClientCreator());
+            // Validate certs.
+            if (iotHubConnectionCredentials.AuthenticationMethod is DeviceAuthenticationWithX509Certificate x509CertificateAuth
+                && x509CertificateAuth.ChainCertificates != null)
+            {
+                if (options.TransportSettings is not IotHubClientAmqpSettings
+                        && options.TransportSettings is not IotHubClientMqttSettings
+                        || options.TransportSettings.Protocol != IotHubClientTransportProtocol.Tcp)
+                {
+                    throw new ArgumentException("Certificate chains are only supported on MQTT over TCP and AMQP over TCP.");
+                }
+            }
+
+            // Make sure client options is initialized.
+            if (options == default)
+            {
+                options = new();
+            }
+
+            InternalClient = new InternalClient(iotHubConnectionCredentials, options, null);
```

Comment:
Next step is to make this a base class?

Reply:
That's a good idea. We should add this to our task list.